src/Services/Chat/Chat.WebAPI/Controllers/ConversationController.cs

```diff
@@ -0,0 +1,87 @@
+﻿using Chat.Application.Features.Conversations.Commands.CreateConversation;
+using Chat.Application.Features.Conversations.Commands.DeleteConversation;
+using Chat.Application.Features.Conversations.Queries.CheckConversationExistence;
+using Chat.Application.Features.Conversations.Queries.GetConversation;
+using Chat.Application.Features.Conversations.Queries.GetUserConversations;
+using Chat.Application.Models.DataTransferObjects.Conversations.Requests;
+using Chat.Application.Models.DataTransferObjects.Conversations.Responses;
+using Chat.Domain.Entities;
+using MediatR;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Chat.WebAPI.Controllers;
+
+[Route("api/[controller]")]
+[ApiController]
+[Authorize]
+public class ConversationController(ISender _sender) 
+    : ControllerBase
+{
+    // GET: api/<ConversationController>
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [HttpGet]
+    public async Task<IActionResult> GetUserConversations(CancellationToken token)
+    {
+        var query = new GetAllUserConversationsQuery();
+        var result = await _sender.Send(query, token);
+
+        return Ok(result);
+    }
+
+    // GET: api/<ConversationController>/<id>
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [HttpGet("{id:guid}")]
+    public async Task<IActionResult> GetConversation([FromRoute] Guid id, CancellationToken token)
+    {
+        var query = new GetConversationQuery(id);
+        var result = await _sender.Send(query, token);
+
+        return Ok(result);
+    }
+
+    // GET: api/<ConversationController>/check-existence
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [HttpGet("check-existence")]
+    public async Task<IActionResult> CheckConversationExistence([FromQuery] CheckConversationExistenceRequest request, CancellationToken token)
+    {
+        var query = new CheckConversationExistenceQuery(request);
+        var result = await _sender.Send(query, token);
+
+        return Ok(result);
```

Comment:
why do you need this endpoint, how are you going to use it from the frontend?

Reply:
I have replaced this Action with GetConversationByItem([FromRoute] Guid itemId, ...) in e4cbfa0 commit.

When the user opens an item (product) tab and presses the "Send Message" button, there is a call to GetConversationByItem([FromRoute] Guid itemId, ...). If there is no such conversation (response 404), then the conversation is created with the Post method, otherwise the conversation is retrieved with GetConversation GetConversation([FromRoute] Guid id, ...).